### ISSUE TITLE
Pump remains suspended with unsuspend_if_no_temp set to true

### ISSUE THREAD

**Issue by tim2000s** [openaps/oref0#1031]
As per the title, I don’t seem to get the pump unsuspended once the temp basal has expired with this preference set. Pump is a 722 WW.

**Comment by scottleibrand** [comment]
Logs please.

**Comment by tim2000s** [comment]
This is my log:
```
Checking deliverAt: 2018-06-09T11:43:48.128Z is within 1m of current time: Sat J
un  9 12:43:48 BST 2018
and that smb-suggested.json is less than 1m old
enact/smb-suggested.json: {"temp":"absolute","bg":130,"tick":"+5","eventualBG":1
96,"insulinReq":2.08,"reservoir":"20.5","deliverAt":"2018-06-09T11:43:48.128Z","
sensitivityRatio":1,"predBGs":{"IOB":[130,133,135,137,139,140,141,142,142,143,14
2,142,141,140,139,138,137,137,136,136,136,135,135,135,135,135,135,135,135,135,13
5,135,135,135,135,135,136,136,136,136,136,137,137,137,137,138],"ZT":[130,127,124
,122,120,118,116,114,113,112,111,111,110,110,110,110,111,112,112],"COB":[130,132
,133,133,133,131,129,128,126,125,123,122,121,120,119,119,118,117,117,116,116,116,116,115,115,115,115,115,115,115,115,115,115,116,116,116,116,116,116,117,117,117,117,118,118,118,118,119],"UAM":[130,133,136,139,142,145,148,151,154,157,159,162,165,168,170,172,175,177,179,181,183,184,186,187,189,190,191,191,192,192,193,193,193,193,193,193,194,194,194,194,194,195,195,195,195,196]},"COB":4,"IOB":0.54,"reason":"COB: 4, Dev: 2.1, BGI: -0.2, ISF: 2.2, CR: 9.18, Target: 4.4, minPredBG 8.9, minGuardBG 7.3, IOBpredBG 7.7, COBpredBG 6.6, UAMpredBG 10.9;  insulinReq 2.08; maxBolus 0.5. Microbolusing 0.5U. no temp, setting 5.25U/hr. ","units":0.5,"duration":30,"rate":5.25}
Temp refreshed: monitor/temp_basal.json: {"duration":0,"rate":0,"temp":"absolute"}
enact/smb-enacted.json: "Rate: 0 Duration: 0"
Temp refreshed: monitor/temp_basal.json: {"duration":0,"rate":0,"temp":"absolute"}
Checking pump status (suspended/bolusing): {"status":"normal","bolusing":false,"suspended":true}
Pump suspended; unsuspend_if_no_temp not enabled in preferences.json: leaving pump suspended
Refreshed pumphistory and meal.json
Temp refreshed: monitor/temp_basal.json: {"duration":0,"rate":0,"temp":"absolute"}
```

This is the preferences:
```
{
        "max_iob": 12,
        "max_daily_safety_multiplier": 5,
        "current_basal_safety_multiplier": 6,
        "autosens_max": 1.9,
        "autosens_min": 0.1,
        "rewind_resets_autosens": true,
        "autosens_adjust_targets": true,
        "adv_target_adjustments": true,
        "maxCOB": 180,
        "override_high_target_with_low": false,
        "skip_neutral_temps": false,
        "bolussnooze_dia_divisor": 7,
        "min_5m_carbimpact": 5,
        "carbratio_adjustmentratio": 1,
        "autotune_isf_adjustmentFraction": 0.5,
        "remainingCarbsCap": 90,
        "enableUAM": true,
        "enableSMB_with_bolus": true,
        "enableSMB_with_COB": true,
        "enableSMB_with_temptarget": true,
        "maxSMBBasalMinutes": 90,
        "curve": "rapid-acting",
        "useCustomPeakTime": false,
        "insulinPeakTime": 75,
        "enableSMB_always":true, 
        "allowSMB_with_high_temptarget":true,
        "enableSMB_after_carbs":true,
        "high_temptarget_raises_sensitivity":true, 
        "low_temptarget_lowers_sensitivity":true,
        "sensitivity_raises_target":true,
        "resistance_lowers_target":true,
        "exercise_mode":true,
        "unsuspend_if_no_temp":true,
        "pushoverGlances": true,
        "MaxUAMSMBBasalMinutes": 60,
        "A52_risk_enable":true
}
```

Just dug through the log file - looks like it's the lack of a space after the colon in `"unsuspend_if_no_temp":true,` that has caused the issue, as adding that space resolves the issue.

On the other hand, `"enableSMB_always":true,` seems to work regardless of the space.

**Comment by scottleibrand** [comment]
@tim2000s When you get a chance can you please test this using the 0.7.0-refactor branch to see if the refactored preferences check code handles this correctly?  It looks like 0.7.0-dev does `grep -iq '"unsuspend_if_no_temp": true' preferences.json` (which requires the space), but 0.7.0-refactor checks the json using `jq`.

**Comment by danamlewis** [comment]
Assuming this has been fixed, @tim2000s with 0.7.0 but please reopen if still an issue or flag in the 0.7.0 PR.
